ros/lib/utils.py

```diff
@@ -80,20 +78,22 @@ def user_data_from_identity(identity):
     return identity['user']
 
 
-def validate_type(value, type_):
+def validate_system(reporter, msg):
```

Comment:
If I understood correctly, by removing `validate_type` method, you are defeating the purpose of it.
With empty string in `is_ros`, this statement `is_ros and cloud_provider is not None` will return True.

Reply:
> `is_ros` is not a string, it is a boolean flag. We only set this field when we have the output of `pmlog_summary` or `ros_config` parsers - [here](https://github.com/RedHatInsights/insights-puptoo/blob/master/src/puptoo/process/profile.py#L568).

Could you please confirm the value type of is_ros from a kafka messages that ros backend received? 
(Example - [here](https://github.com/RedHatInsights/ros-backend/blob/main/tests/data_files/events-message.json#L1152) which i am referring to)

Reply:
> > `is_ros` is not a string, it is a boolean flag. We only set this field when we have the output of `pmlog_summary` or `ros_config` parsers - [here](https://github.com/RedHatInsights/insights-puptoo/blob/master/src/puptoo/process/profile.py#L568).
> 
> @r14chandra - Could you please confirm the value type of is_ros from a kafka messages that ros backend received? (Example - [here](https://github.com/RedHatInsights/ros-backend/blob/main/tests/data_files/events-message.json#L1152) which i am referring to)

We are adding `is_ros` as a boolean in [system_profile](https://github.com/RedHatInsights/insights-puptoo/blob/master/src/puptoo/process/profile.py#L569) but in platform_metadata, we are explicitly making it string - [here](https://github.com/RedHatInsights/insights-puptoo/blob/master/src/puptoo/app.py#L240-L241).
IMO we should consider making this field a boolean type to make it consistent.

@kgaikwad - To answer your query, the condition would anyway return false for the cases.

Reply:
> @kgaikwad - To answer your query, the condition would anyway return false for the cases.

My bad! I evaluated result using brackets i.e (is_ros and cloud_provide) 

@upadhyeammit @r14chandra  - I still think, we should prefer to use type casting (i.e `validate_type` on `is_ros`) on backend side. Thoughts?

Reply:
I had discussion with @r14chandra and as per that its Puptoo which has inconsistency while generating the platform_metadata, so we should fix it and return True from there for is_ros. This means when we parse it back at ROS side it will be either True or None. So accordingly we can skip the typecasting.

Reply:
@r14chandra and @upadhyeammit,  +1 for fixing the value under platform_metadata on Puptoo side.
 
@r14chandra -  could you please raise the issue on puptoo side and mark this conversation resolve?

Reply:
Here is the card - [ESSNTL-4611](https://issues.redhat.com/browse/ESSNTL-4611)

Reply:
This PR is required to merge before [insights-puptoo-#269](https://github.com/RedHatInsights/insights-puptoo/pull/269)
as the current ROS stage contains [this](https://github.com/RedHatInsights/ros-backend/blob/main/ros/lib/utils.py#L93) conditions.